cpp/src/cider/exec/nextgen/jitlib/llvmjit/LLVMJITEngine.cpp

```diff
@@ -67,10 +71,10 @@ std::unique_ptr<LLVMJITEngine> LLVMJITEngineBuilder::build() {
   engine->engine->DisableLazyCompilation(false);
   engine->engine->setVerifyModules(false);
 
-  engine->engine->RegisterJITEventListener(
-      llvm::JITEventListener::createPerfJITEventListener());
-  engine->engine->RegisterJITEventListener(
-      llvm::JITEventListener::createIntelJITEventListener());
+  engine->perf_listener = llvm::JITEventListener::createPerfJITEventListener();
+  engine->engine->RegisterJITEventListener(engine->perf_listener);
+  engine->intel_listener = llvm::JITEventListener::createIntelJITEventListener();
+  engine->engine->RegisterJITEventListener(engine->intel_listener);
```

Comment:
any benefits?

Reply:
The listeners need to be released explicitly, LLVM engine will not handle it.

Reply:
but If we don't pass the argument， It just gets the pointer from a static variable， which is no need to release explicitly.

Reply:
But sanitizer will throw memleak if there is no release.

Reply:
I think it's a sanitizer flaw， we should not add overhead just to make the sanitizer happy.
any method to make sanitizer ignore these pointers?